From c7d56c3e7bf6616b7d6fda1fbe247be82b5a0537 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Mon, 12 Mar 2012 16:59:51 +0000
Subject: [PATCH 505/505] alsa-mixer: Allow speakers even when no specific
 path is available.

Several laptops (e.g. mine) have built in speakers and a headphone jack
but do not specifically provide a separate control element in alsa
to adjust speaker vs. headphone volumes (i.e. both are controlled by
'Master').

In this case, there is no way to show volumes separately and eventually
when routing information takes availability information into account
it might even cause routing problems.

To deal with this we ensure both a Speakers and a Headphones port
are added. This will likely cause some phantom Speakers port
usage on desktop machines, but for now this is the best setup.

Longer term, we can perhaps use form factor or some other (kernel-side)
metadata to avoid this case
---
 src/modules/alsa/mixer/paths/analog-output-speaker.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/modules/alsa/mixer/paths/analog-output-speaker.conf b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
index 827fcfd..af0ae46 100644
--- a/src/modules/alsa/mixer/paths/analog-output-speaker.conf
+++ b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
@@ -25,6 +25,7 @@ name = analog-output-speaker
 [Jack Headphone]
 state.plugged = no
 state.unplugged = unknown
+required-any = any
 
 [Jack Front Headphone]
 state.plugged = no
-- 
1.7.11.3

